### PR TITLE
Implemented ``unused-private-member`` checker

### LINKED ISSUE #4483: Add check for unused private members
Pylint should check if a private class member (started with `__`) is used at least once after definition. My case - private functions left after refactoring.

### PR DESCRIPTION
**Pull request by yushao2**
## Steps

- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description
Implemented ``unused-private-member`` checker, which is emitted when a private member of a class is defined but unused

Checks for:
- Private class variables
- Private instance attributes
- Private methods

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Related Issue

Closes #4483

### CODE REVIEW

File: tests/functional/u/unused/unused_private_member.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    def __test(self, x, y, z):  # [unused-private-member]
+        fn = HasUnusedInClass.__private_class_method_used
+        fn()
+        fn2 = HasUnusedInClass.__private_static_method_used
+        fn2()
+
+    def __my_print(self, string):
+        print(self.__another_secret + string)
+        another_obj = AnotherClass()
+        another_obj.__test()  # this class's test should still be unused
```
Comment (Pierre-Sassoulas) [inline tests/functional/u/unused/unused_private_member.py:43]: 👍

File: pylint/checkers/classes.py [hunk 2]
```diff
@@ -636,6 +636,11 @@ def _has_same_layout_slots(slots, assigned_value):
         "Used when a method parameter has a different name than in "
         "the implemented interface or in an overridden method.",
     ),
+    "W0238": (
+        "Unused private member of class %s: %s",
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/classes.py:640]:
```suggestion
        "Unused private member '%s.%s'",
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you for this new checker, it will permit to clean up a lot of unused code in badly tested code bases !

**Comment by cdce8p** [comment]
I found a false positive. It's not that common for python, but at times used for the singleton pattern.
```py
class MyCls:
    __class_var = None

    @classmethod
    def set_class_var(cls, var):
        cls.__class_var = var  # false-positive: unused-private-member

    @classmethod
    def get_class_var(cls):
        return cls.__class_var
```

**Comment by yushao2** [comment]
> I found a false positive. It's not that common for python, but at times used for the singleton pattern.
> 
> ```python
> class MyCls:
>     __class_var = None
> 
>     @classmethod
>     def set_class_var(cls, var):
>         cls.__class_var = var  # false-positive: unused-private-member
> 
>     @classmethod
>     def get_class_var(cls):
>         return cls.__class_var
> ```

I'll look into this

**Pull request merged**